core/src/main/java/org/infinispan/AdvancedCache.java

```diff
@@ -559,6 +572,36 @@
    CacheSet<CacheEntry<K, V>> cacheEntrySet();
 
    /**
+    * Returns a sequential stream using this Cache as the source. This stream is very similar to using the
+    * {@link CacheStream} returned from the {@link CacheSet#stream()} method of the collection
+    * returned via {@link AdvancedCache#cacheEntrySet()}. The use of this locked stream is that when an entry is
+    * being processed by the user the entry is locked for the invocation preventing a different thread from modifying
+    * it.
+    * <p>
+    * Note that this stream is not supported when using a optimistic transactional or simple cache. Both non transactional and
+    * pessimistc transactions are supported.
+    * <p>
+    * The stream will not share any ongoing transaction the user may have. Code executed by the stream should be treated
+    * as completely independent. That is any operation performed via the stream will require the user to start their
+    * own transaction or will be done intrinsically on the invocation. Note that if there is an ongoing transaction
+    * that has a lock on a key from the cache, that it will cause a deadlock.
+    * <p>
+    * Currently simple cache, {@link org.infinispan.configuration.cache.ConfigurationBuilder#simpleCache(boolean)}
+    * was set to true, functional map {@link org.infinispan.commons.api.functional.FunctionalMap} and optimistic caches,
```

Comment:
`org.infinispan.commons.api.functional.FunctionalMap` is not an AdvancedCache, so it can't support this method. You have mistaken it for `org.infinispan.functional.decorators.FunctionalAdvancedCache` which is a test class?